plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php

```diff
@@ -15,10 +15,23 @@ class WC_Beta_Tester_Live_Branches {
 	 * Constructor.
 	 */
 	public function __construct() {
+		if ( ! $this->woocommerce_is_installed() ) {
+			return;
+		}
+
 		add_action( 'admin_menu', array( $this, 'register_page' ) );
 		add_action( 'admin_init', array( $this, 'register_scripts' ) );
 	}
 
+	/**
+	 * Check if WooCommerce is installed.
+	 *
+	 * @return bool - True if WooCommerce is installed, false otherwise.
+	 */
+	private function woocommerce_is_installed() {
+		return class_exists( 'WooCommerce' );
```

Comment:
If this check wasn't sufficient in the case of `plugins/woocommerce-beta-tester/api/tracks/class-tracks-debug-log.php`, why not just directly make this a check for `defined( 'WC_ABSPATH' )`?

Reply:
@psealock the log code has that flaw because it asks if woocommerce class exists then tries to access a constant defined long after the class exists. You can see how the order of checks is wrong. That class needs refactoring or the constant loaded earlier. We can refactor at some point. I would rather not have to check the constant exists because those constants are imo more of a private api that could change.

This code doesnt have that flaw. It doesnt use that late defined constant and checking if woocommerce class exists is in 99% of cases the more sensible check in my opinion. This code will never break in that way because if woo doesnt exist we never do anything. 

Tldr; one is a patch to an issue i dont want to refactor immediately. the other is doing things the more proper way imo.